Split/FetcherEngine/Refresh/RetryableSyncWorker.swift

```diff
@@ -101,15 +108,21 @@ class RetryableMySegmentsSyncWorker: BaseRetryableSyncWorker {
         self.mySegmentsStorage = mySegmentsStorage
         self.mySegmentsFetcher = mySegmentsFetcher
         self.metricsManager = metricsManager
+        self.changeChecker = DefaultMySegmentsChangesChecker()
+
         super.init(eventsManager: eventsManager, reconnectBackoffCounter: reconnectBackoffCounter)
     }
 
     override func fetchFromRemote() -> Bool {
         do {
+            let oldSegments = mySegmentsStorage.getAll()
             if let segments = try self.mySegmentsFetcher.execute(userKey: self.userKey) {
                 Logger.d(segments.debugDescription)
-                mySegmentsStorage.set(segments)
-                fireReadyIsNeeded(event: SplitInternalEvent.mySegmentsAreReady)
+                if !isSdkReadyTriggered() ||
```

Comment:
Javi, check if you really need the `isSdkReadyTriggered` condition, because, at least in JS, it doesn't need to check if the SDK_READY event has been emitted or not. We call `notifyMySegmentsUpdated()` anyway.